src/lib.rs

```diff
@@ -186,7 +271,8 @@ impl Connection<NotConnected> {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn new() -> Connection<NotConnected> {
+    #[allow(clippy::new_without_default)]
+    pub const fn new() -> Connection<NotConnected> {
```

Comment:
`const fn` is even better than `inline(always)` because it gets completely evaluated at compile time.